Result/ResultType.swift

```diff
@@ -20,12 +20,12 @@ public protocol ResultType {
 public extension ResultType {
 	
 	/// Returns the value if self represents a success, `nil` otherwise.
-	var value: Value? {
+	public var value: Value? {
 		return analysis(ifSuccess: { $0 }, ifFailure: { _ in nil })
 	}
 	
 	/// Returns the error if self represents a failure, `nil` otherwise.
-	var error: Error? {
+	public var error: Error? {
```

Comment:
I’ve been wondering if these should be declared in `ResultType` as well. Default implementations would be provided by this extension, but declaring them in `ResultType` would allow conforming types to provide their own implementations, which could be useful for efficiency or debugging.